### PR TITLE
[DEVOPS-1006] Configure source filter to apply to all plutus packages

### PR DESCRIPTION
**Pull request by disassembler**

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -31,32 +31,30 @@ let
...
       lib.hasSuffix ".dhall" baseName ||
-      # Filter out nix-build result symlinks
       (type == "symlink" && lib.hasPrefix "result" baseName) ||
       (type == "directory" && baseName == ".stack-work")
     );
-  source = builtins.filterSource cleanSourceFilter ./.;
+  requiredOverlay = self: super: {
+    plutus-prototype = addRealTimeTestLogs super.plutus-prototype;
+    language-plutus-core = doHaddockHydra (addRealTimeTestLogs super.language-plutus-core);
+    mkDerivation = args: super.mkDerivation (args // optionalAttrs (args ? src) {
```
Comment (michaelpj) [inline default.nix]: This feels a little aggressive to me, but I guess it makes sense given the limited package set we have here. I'd probably have just put it in another function, like `doHaddockHydra` and friends.

Comment (michaelpj) [reply default.nix]: Also, please comment for posterity on why these things are necessary, e.g. "Ensure that all Haskell packages we build will have their source filtered"

Comment (michaelpj) [reply default.nix]: Actually, thinking about it, won't this also apply to all the packages in the generated package set? Are we sure it's safe for all of them?

**Comment by michaelpj** [comment]
ping - would be good to merge this

**Comment by mchakravarty** [comment]
Why do the CI jobs not complete?

**Comment by michaelpj** [comment]
(To be clear, I would like something to happen about my comments)

**Pull request merged**